### PR TITLE
Login: Display inline validation messages in `LoginForm`

### PR DESCRIPTION
**Pull request by drewblaisdell**
screen shot 2017-05-05 at 4 35 26 pm

This PR:

- Updates `LoginForm` to use properties from the global store instead of properties from `this.state` when possible.
- Updates the selectors/tests for `login.requestError` in the global store to use an object instead of a string.
- Updates `LoginForm` to display inline validation messages.

**Testing**
- Visit `/login`.
- Enter an invalid username/password combination.
- Assert that you see an inline error message like above.

- [x] Code
- [x] Product

### CODE REVIEW

**Comment by hafizrahman** [comment]
I might be missing something obvious here, but the error message is not displaying for me.

Here's a video:

https://cloudup.com/cB5eUszcbWw

**Comment by gziolo** [comment]
@hafizrahman you need to be proxied :)

File: client/login/wp-login/index.jsx [hunk 1]
```diff
@@ -138,6 +140,8 @@ class Login extends React.Component {
 		return (
 			<Main className="wp-login">
 				<PageViewTracker path="/login" title="Login" />
+				<GlobalNotices id="notices" notices={ notices.list } />
```
Comment (gziolo) [inline client/login/wp-login/index.jsx]: It probably should land in the logged out layout component, but let's keep it here for the time being.

**Comment by gziolo** [comment]
We probably should update server side code to display different messages for invalid username && password pair. This is a bit confusing:

screen shot 2017-05-08 at 09 34 55
screen shot 2017-05-08 at 09 35 05

**Comment by Tug** [comment]
Yep I don't think we should even try to show inline messages for errors returned by `loginUser` endpoint, we know the server isn't supposed to give away too much information.
How about we keep a global notice for that case?

**Comment by gziolo** [comment]
Yes, it works, too :)

**Comment by drewblaisdell** [comment]
> Yep I don't think we should even try to show inline messages for errors returned by loginUser endpoint, we know the server isn't supposed to give away too much information.
How about we keep a global notice for that case?

We don't return any data that isn't currently returned by the existing login, so I think we can just change the messages to be more specific.

File: client/state/login/actions.js [hunk 2]
```diff
@@ -22,11 +22,22 @@ const loginErrorMessages = {
 	empty_password: translate( 'The password field is empty.' ),
 	empty_two_step_code: translate( 'The verification code field is empty.' ),
 	empty_username: translate( 'The username field is empty.' ),
-	incorrect_password: translate( 'Invalid username or password.' ),
+	incorrect_password: translate( 'Invalid password.' ),
 	invalid_two_step_code: translate( 'Invalid verification code.' ),
-	invalid_username: translate( 'Invalid username or password.' ),
+	invalid_username: translate( 'Invalid username.' ),
```
Comment (a8ci18n) [inline client/state/login/actions.js]: :disappointed: 40 existing translations will be lost with this change.
If the string must change, the following option has already been translated **1** time:
`translate( 'Invalid username!' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/login-validation)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/state/login/actions.js [hunk 3]
```diff
@@ -22,11 +22,22 @@ const loginErrorMessages = {
 	empty_password: translate( 'The password field is empty.' ),
 	empty_two_step_code: translate( 'The verification code field is empty.' ),
 	empty_username: translate( 'The username field is empty.' ),
-	incorrect_password: translate( 'Invalid username or password.' ),
+	incorrect_password: translate( 'Invalid password.' ),
```
Comment (a8ci18n) [inline client/state/login/actions.js]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Invalid password' )` *ES Score: 17*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/login-validation)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (yoavf) [inline client/state/login/actions.js]: Or you can add the exclamation mark here too

**Comment by hafizrahman** [comment]
Product LGTM

**Comment by Tug** [comment]
Rebased and added a simple case.

**Pull request merged**